File: app/src/main/java/com/github/ykiselev/playground/init/ErrorCallbackBootstrap.java
package com.github.ykiselev.playground.init;

import com.github.ykiselev.common.closeables.Closeables;
import org.lwjgl.glfw.GLFWErrorCallback;

import static org.lwjgl.glfw.GLFW.glfwSetErrorCallback;

public final class ErrorCallbackBootstrap implements AutoCloseable {

    private final GLFWErrorCallback callback;

    private final GLFWErrorCallback previous;

    public ErrorCallbackBootstrap() {
        this.callback = GLFWErrorCallback.createPrint(System.err);
        this.previous = glfwSetErrorCallback(callback);
    }

    @Override
    public void close() {
        glfwSetErrorCallback(previous);
        Closeables.close(callback);
    }
}

File: app/src/main/kotlin/com/github/ykiselev/playground/init/ErrorCallbackBootstrap.kt
package com.github.ykiselev.playground.init

import com.github.ykiselev.common.closeables.Closeables
import org.lwjgl.glfw.GLFW
import org.lwjgl.glfw.GLFWErrorCallback

class ErrorCallbackBootstrap : AutoCloseable {

    private val callback: GLFWErrorCallback = GLFWErrorCallback.createPrint(System.err)
    private val previous: GLFWErrorCallback? = GLFW.glfwSetErrorCallback(callback)

    override fun close() {
        GLFW.glfwSetErrorCallback(previous)
        Closeables.close(callback)
    }
}